File: app/src/main/java/linusfessler/alarmtiles/alarm/Alarm.java
package linusfessler.alarmtiles.alarm;

import androidx.room.Entity;
import androidx.room.PrimaryKey;

import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.ToString;

@Data
@Entity
@ToString
@EqualsAndHashCode
public class Alarm {

    @PrimaryKey(autoGenerate = true)
    private Long id;

    private boolean enabled;
    private int hourOfDay;
    private int minuteOfHour;

    // TODO: Add a constructor with sensible default values
    public Alarm() {
        this.hourOfDay = 7;
        this.minuteOfHour = 0;
    }

    void toggle() {
        if (this.isEnabled()) {
            this.disable();
        } else {
            this.enable();
        }
    }

    void enable() {
        this.setEnabled(true);
    }

    void disable() {
        this.setEnabled(false);
    }
}

File: app/src/main/java/linusfessler/alarmtiles/alarm/Alarm.kt
package linusfessler.alarmtiles.alarm

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity
data class Alarm(
        @field:PrimaryKey val id: Long = 0,
        val isEnabled: Boolean = false,
        val hourOfDay: Int = 7,
        val minuteOfHour: Int = 0,
        val triggerTimestamp: Long? = null
) {
    fun enable(hourOfDay: Int, minuteOfHour: Int, triggerTimestamp: Long): Alarm {
        return copy(isEnabled = true, hourOfDay = hourOfDay, minuteOfHour = minuteOfHour, triggerTimestamp = triggerTimestamp)
    }

    fun disable(): Alarm {
        return copy(isEnabled = false, triggerTimestamp = null)
    }
}